python/ray/train/v2/_internal/execution/controller/controller.py

```diff
@@ -219,13 +222,16 @@ def _execute_failure_decision(
         training_failed_error = TrainingFailedError(
             error_message=errors_str, worker_failures=worker_group_status.errors
         )
-
-        if failure_decision == FailureDecision.RESTART:
-            logger.error(
-                "Restarting training worker group after encountering "
-                f"failures on {len(worker_group_status.errors)} worker(s):\n"
-                f"{errors_str}"
+        if failure_decision == FailureDecision.RESCHEDULE:
+            assert isinstance(worker_group_status, WorkerGroupResizeStatus)
+            return TrainControllerLoopIterationResult(
+                run_attempt_id=self._get_run_attempt_id(),
+                previous_state=controller_state,
+                next_state=ReschedulingState(),
             )
+
+        elif failure_decision == FailureDecision.RESTART:
+            logger.error(worker_group_status.get_restart_error_string())
```

Comment:
what about merging `RESCHEDULE` and `RESTART` failure decisions --> `RETRY`

Then, if we decide to RETRY, pick the next state based on the current state: if we're currently RUNNING, then go to RESTARTING. If we're currently SCHEDULING, then go to RESCHEDULING